### DISCUSSION TITLE
Why does version update occur after loading of pool and balance modules?

### DISCUSSION THREAD

**Discussion by AnthonyBe** [RainbowMiner/RainbowMiner#1689] [Q&A] [answered]
I've been meaning to ask for a while now. When an update is detected on initial startup, why wait until _after_ loading pool modules and balance modules, which can often take many minutes, before downloading and applying the update? Why is the update not the first thing to process?

```
Starting v4.7.0.8! Please wait..

Checking for VM .. ok (not in a VM)
Checking for cURL .. ok (disabled)
Detecting devices .. 1 CPU, 1 Nvidia found
Checking Windows pagefile/virtual memory .. ok
Initialize configuration ..
Read configuration ..
WARNING: RainbowMiner is out of date: lastest release version v4.7.0.9 is available.
Loading pool modules ..
.. loading unMineable done (4.635s)
Loading balance modules ..
```

**Comment by RainbowMiner** [comment, marked as answer]
It has something to do with automatic error correction, in case something bad happens during an update (e.g. internet goes down, power down etc.). It avoids running into a potential no-operation cycle - it's a tradeoff.
